src/network/networkd-netdev-wireguard.c

```diff
@@ -0,0 +1,625 @@
+                                   lvalue,
+                                   ltype,
+                                   rvalue,
+                                   data,
+                                   &w->dev.preshared_key);
+}
+
+
+int config_parse_wireguard_public_key(const char *unit,
+                                      const char *filename,
+                                      unsigned line,
+                                      const char *section,
+                                      unsigned section_line,
+                                      const char *lvalue,
+                                      int ltype,
+                                      const char *rvalue,
+                                      void *data,
+                                      void *userdata) {
+        Wireguard *w;
+        WireguardPeer *peer;
+
+        assert(data);
+
+        w = WIREGUARD(data);
+
+        assert(w);
+
+        peer = new_peer(w, section_line);
+        if (!peer)
+                return -ENOMEM;
+
+        return parse_wireguard_key(unit,
+                                   filename,
+                                   line,
+                                   section,
+                                   section_line,
+                                   lvalue,
+                                   ltype,
+                                   rvalue,
+                                   data,
+                                   peer->fields.public_key);
+}
+
+int config_parse_wireguard_allowed_ips(const char *unit,
+                                       const char *filename,
+                                       unsigned line,
+                                       const char *section,
+                                       unsigned section_line,
+                                       const char *lvalue,
+                                       int ltype,
+                                       const char *rvalue,
+                                       void *data,
+                                       void *userdata) {
+        _cleanup_strv_free_ char **words = NULL;
+        char **word;
+        const char *address, *end;
+        union in_addr_union buffer;
+        unsigned char prefixlen;
+        int r, family;
+        Wireguard *w;
+        WireguardPeer *peer;
+        WireguardIpmask *ipmask;
+
+        assert(rvalue);
+        assert(data);
+
+        w = WIREGUARD(data);
+
+        peer = new_peer(w, section_line);
+        if (!peer)
+                return -ENOMEM;
+
+        r = strv_split_extract(&words, rvalue, ",", 0);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to split allowed ips \"%s\" option: %m", rvalue);
+                return 0;
+        }
+        STRV_FOREACH(word, words) {
+                end = strchr(*word, '/');
+                if (end)
+                        address = strndupa(*word, end - *word);
```

Comment:
it's not OK to use strndupa() within a loop, as it allocates memory on the stack. See CODING_STYLE.